containers/init/clone/clone.sh

```diff
@@ -14,8 +14,11 @@
 # limitations under the License.
 
 set -ex
-
 cd /src/workspace
 ls -A | xargs -r rm -fr
-git clone --recursive $CLONE_REPO .
-git checkout $CLONE_GIT_REF
+git init
+git remote add origin $CLONE_REPO
+git fetch
+git reset --hard $CLONE_GIT_REF
+git submodule update --init --recursive
+chmod -R 777 .
```

Comment:
why is the chmod required?

Reply:
I don't make assumptions that the container will only be used with kubernetes. In fact, I test containers using docker alone before I ever throw them at kubernetes. That was the case here.

When mounting a pure docker volume and running this container, the resulting repository has inaccessible permissions. The chmod ensures that the contents written to the volume are mutable on both ends of the volume.

In kubernetes, immutability with volumes will be reintroduced soon as a property of the Volume type. It is currently an alpha feature. By default, kubernetes allows volumes to be read/write. This masks the problem under the hood.

Reply:
The concern is that git repos actually persist some mode information along with the file (e.g. the executable flag for scripts). So it feels wrong that you check out a repo and immediately overwrite all the information provided by git.
(e.g. non-executable scripts are changed into executable scripts which can lead to a different behavior when building locally vs when building in kubernetes - imagine someone forgets to set an executable flag for their script but things magically work in the oss benchmarks). At the very least you should provide a comment why the `chmod -R 777 .` is useful here.